public/js/pimcore/helpers.js

```diff
@@ -2936,6 +2936,34 @@ pimcore.helpers.isValidPassword = function (pass) {
     return true;
 };
 
+pimcore.helpers.isComplexPassword = function (pass) {
+    if (pass.length < 8 || pass.length > 12) {
+        return false;
+    }
+
+    const uppercase = /[A-Z]/.test(pass);
+    const lowercase = /[a-z]/.test(pass);
+    const numbers = /d/.test(pass);
+    const specialCharacters = /[^\w]/.test(pass);
+
+    return !(!uppercase || !lowercase || !numbers || !specialCharacters);
+}
+
+pimcore.helpers.isLongLessComplexPassword = function (pass) {
+    if (pass.length < 25) {
+        return false;
+    }
+
+    const uppercase = /[A-Z]/.test(pass);
+    const lowercase = /[a-z]/.test(pass);
```

Comment:
These expressions will not detect lower/upper characters like `åÅ` and `øØ` and `üÜ`.

Reply:
Yes, the function in line 2960 is responsible for special characters.
`const specialCharacters = /[^\w]/.test(pass);`

Reply:
The two functions are only responsible for normal letters. That is why I have also included the other test for special letters. I don't think the BSI counts the characters as capital letters. I would have to read that again in more detail.

Reply:
If it's part of the alphabet it's not a special character.

Reply:
I think you mean the German ö ä ü. Then yes, they look funny here in github. I have added them now sorry for the incomprehensibility

Reply:
I meant pretty much any Unicode letter which has an uppercase and lowercase variant, in addition to the ones you added that also includes European letters such as [Øø](https://en.wikipedia.org/wiki/%C3%98), [Ææ](https://en.wikipedia.org/wiki/%C3%86) [ẞß](https://en.wikipedia.org/wiki/%C3%9F), [Çç](https://en.wikipedia.org/wiki/%C3%87)…

Try them using Unicode instead of adding them one by one in the test: 

```js
/\p{Lowercase_Letter}/u.test(pass);
```

Reply:
Okay, I have now changed the regex statement.